bot/kodiak/evaluation.py

```diff
@@ -96,27 +136,57 @@ def get_merge_body(config: V1, pull_request: PullRequest) -> MergeBody:
             merge_body.commit_message = pull_request.url
         else:
             merge_body.commit_message += "\n\n" + pull_request.url
+
+    co_author_trailers = []
     if config.merge.message.include_pull_request_author:
-        commit_message = (
-            merge_body.commit_message if merge_body.commit_message is not None else ""
+        author = get_commit_author_info(
+            login=pull_request.author.login,
+            databaseId=pull_request.author.databaseId,
+            name=pull_request.author.name,
+            type_=pull_request.author.type,
         )
-        author_name = pull_request.author.login
-        author_login = pull_request.author.login
-        if pull_request.author.type == "Bot":
-            author_name += "[bot]"
-            author_login += "[bot]"
-        if pull_request.author.name:
-            author_name = pull_request.author.name
-
-        # GitHub does not allow our GitHub App to view the email addresses of
-        # pull request authors, so we generate a noreply GitHub email address
-        # instead which works the same for the GitHub UI.
-        author_email = (
-            f"{pull_request.author.databaseId}+{author_login}@users.noreply.github.com"
+        co_author_trailers.append(
+            get_coauthor_trailer(
+                user_id=pull_request.author.databaseId,
+                login=author.login,
+                name=author.name,
+            )
         )
+    if config.merge.message.include_coauthors:
+        for commit_author in commit_authors:
+            if (
+                commit_author.databaseId is None
+                # don't add trailers for pull request author.
+                # TODO(chdsbd): Should we remove this?
+                or commit_author.databaseId == pull_request.author.databaseId
+            ):
+                continue
+            author = get_commit_author_info(
+                login=commit_author.login,
+                databaseId=commit_author.databaseId,
+                name=commit_author.name,
+                type_=commit_author.type,
+            )
+
+            co_author_trailers.append(
+                get_coauthor_trailer(
+                    user_id=commit_author.databaseId,
+                    login=author.login,
+                    name=author.name,
+                )
+            )
+
+    if co_author_trailers and config.merge.message.body not in (
+        MergeBodyStyle.empty,
+        MergeBodyStyle.github_default,
+    ):
```

Comment:
should this be a whitelist instead of a blacklist? Such that a new MergeBodyStyle wouldn't automatically get the coauthors?